test/modules/EdgeHubRestartTester/src/Settings.cs

```diff
@@ -0,0 +1,150 @@
+    {
+        internal static Settings Current = Create();
+
+        Settings(
+            TimeSpan sdkRetryTimeout,
+            string serviceClientConnectionString,
+            string deviceId,
+            string reportingEndpointUrl,
+            TimeSpan restartPeriod,
+            TimeSpan testStartDelay,
+            TimeSpan testDuration,
+            bool directMethodEnable,
+            string directMethodName,
+            string directMethodTargetModuleId,
+            bool messageEnable,
+            string messageOutputEndpoint,
+            TransportType transportType,
+            string moduleId,
+            string trackingId)
+        {
+            Preconditions.CheckRange(sdkRetryTimeout.Ticks, 0);
+            Preconditions.CheckRange(restartPeriod.Ticks, 0);
+            Preconditions.CheckRange(testStartDelay.Ticks, 0);
+            Preconditions.CheckRange(testDuration.Ticks, 0);
+
+            this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.DirectMethodEnable = Preconditions.CheckNotNull(directMethodEnable, nameof(directMethodEnable));
+            this.DirectMethodName = this.DirectMethodEnable ? Preconditions.CheckNonWhiteSpace(directMethodName, nameof(directMethodName)) : directMethodName;
+            this.DirectMethodTargetModuleId = this.DirectMethodEnable ? Preconditions.CheckNonWhiteSpace(directMethodTargetModuleId, nameof(directMethodTargetModuleId)) : directMethodTargetModuleId;
+            this.MessageEnable = Preconditions.CheckNotNull(messageEnable, nameof(messageEnable));
+            this.MessageOutputEndpoint = this.MessageEnable ? Preconditions.CheckNonWhiteSpace(messageOutputEndpoint, nameof(messageOutputEndpoint)) : messageOutputEndpoint;
+            this.TransportType = transportType;
+            this.ModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            this.ReportingEndpointUrl = new Uri(Preconditions.CheckNonWhiteSpace(reportingEndpointUrl, nameof(reportingEndpointUrl)));
+            this.RestartPeriod = restartPeriod;
+            this.SdkRetryTimeout = (uint)sdkRetryTimeout.Milliseconds;
+            this.ServiceClientConnectionString = Preconditions.CheckNonWhiteSpace(serviceClientConnectionString, nameof(serviceClientConnectionString));
+            this.TestDuration = testDuration;
+            this.TestStartDelay = testStartDelay;
+            this.TrackingId = trackingId;
+
+            if (!(this.DirectMethodEnable || this.MessageEnable))
+            {
+                throw new NotSupportedException("EdgeHubRestartTester requires at least one of the sending methods {DirectMethodEnable, MessageEnable} to be enable to perform the EdgeHub restarting test.");
+            }
+
+            if (restartPeriod < sdkRetryTimeout)
+            {
+                throw new InvalidDataException("sdkRetryTimeout period must be less than restartInterval period {sdkRetryTimeoutMilisec < restartPeriod}.");
```

Comment:
why print "{sdkRetryTimeoutMilisec < restartPeriod}"?

Reply:
I meant to be very explicit.  I can remove it as requested.